node/src/chain/fsm.rs

```diff
@@ -505,168 +506,163 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> InSyncImpl<DB, VM, N> {
-                {
-                    let local_height = prev_header.height + 1;
-                    if let Some(l_b) = t.fetch_block_by_height(local_height)? {
-                        if remote_blk.header().iteration
-                            < l_b.header().iteration
-                        {
-                            return Ok(Some((
-                                l_b.header().clone(),
-                                prev_header.state_hash,
-                            )));
-                        }
-                    }
+            // Ensure the block is different from what we have in our chain
+            if remote_height == tip_header.height {
+                if remote_header.hash == tip_header.hash {
+                    return Ok(None);
                 }
-
-                anyhow::Ok(None)
-            })?;
-
-            if let Some((local_header, state_hash)) = result {
-                match fallback::WithContext::new(acc.deref())
-                    .try_revert(
-                        &local_header,
-                        remote_blk.header(),
-                        RevertTarget::Commit(state_hash),
-                    )
+            } else {
+                let blk_exists = acc
+                    .db
+                    .read()
                     .await
-                {
-                    Ok(_) => {
-                        counter!("dusk_fallback_count").increment(1);
-                        if remote_height == acc.get_curr_height().await + 1 {
-                            acc.try_accept_block(remote_blk, true).await?;
-                            return Ok(None);
-                        }
-                    }
-                    Err(e) => {
-                        error!(
-                            event = "fallback failed",
-                            height = local_header.height,
-                            remote_height,
-                            err = format!("{:?}", e)
-                        );
-                        return Ok(None);
-                    }
+                    .view(|t| t.get_block_exists(&remote_header.hash))?;
+
+                if blk_exists {
+                    return Ok(None);
                 }
             }
 
-            return Ok(None);
-        }
+            // Check if prev_blk is in our chain
+            // If not, remote_blk is on a fork
+            let prev_blk_exists =
+                acc.db.read().await.view(|t| {
+                    t.get_block_exists(&remote_header.prev_block_hash)
+                })?;
 
-        if remote_height == local_header.height {
-            if remote_blk.header().hash == local_header.hash {
-                // Duplicated block.
-                // Node has already accepted it.
+            if !prev_blk_exists {
+                warn!(
+                    "received block from fork at height {remote_height}: {}",
+                    to_str(&remote_header.hash)
+                );
```

Comment:
So what should we do here? Should we send all our non final blocks to this guy? (otherwise we'll never get in sync)

Reply:
This should be taken care of in another dedicated PR

Reply:
This will be handled by Issue #2462